src-opam/dockerfile_distro.ml

```diff
@@ -291,17 +291,19 @@ let tag_of_distro (d:t) = match d with
   |`OpenSUSE `V15_1 -> "opensuse-15.1"
   |`OpenSUSE `V15_2 -> "opensuse-15.2"
   |`OpenSUSE `Latest -> "opensuse"
-  |`Cygwin `V20H2 -> "cygwin-20H2"
+  |`Cygwin `V21H1 -> "cygwin-21H1"
   |`Windows (`Mingw, `V1809) -> "windows-mingw-1809"
   |`Windows (`Mingw, `V1903) -> "windows-mingw-1903"
   |`Windows (`Mingw, `V1909) -> "windows-mingw-1909"
   |`Windows (`Mingw, `V2004) -> "windows-mingw-2004"
   |`Windows (`Mingw, `V20H2) -> "windows-mingw-20H2"
+  |`Windows (`Mingw, `V21H1) -> "windows-mingw-21H1"
   |`Windows (`Msvc, `V1809) -> "windows-msvc-1809"
   |`Windows (`Msvc, `V1903) -> "windows-msvc-1903"
   |`Windows (`Msvc, `V1909) -> "windows-msvc-1909"
   |`Windows (`Msvc, `V2004) -> "windows-msvc-2004"
   |`Windows (`Msvc, `V20H2) -> "windows-msvc-20H2"
+  |`Windows (`Msvc, `V21H1) -> "windows-msvc-21H1"
```

Comment:
A similar refactoring with a function to convert the version I think is worth doing for safety, given that this will be changed every six months (and Microsoft change the naming frequently enough that I don't think we should generate the names!)